dss/operations/checkout.py

```diff
@@ -0,0 +1,183 @@
+
+
+def verify_blob_existance(handle, bucket, key):
+    """Helper Functiont to see if blob data exists"""
+    try:
+        file_metadata = handle.get(bucket=bucket, key=key)
+    except BlobNotFoundError:
+        return False
+    return True
+
+
+def sleepy_checkout(function=None, **kwargs):
+    status = False
+    token = None
+    key = kwargs["file_metadata"]['name'] if kwargs['file_metadata'] else kwargs.get("bundle_uuid")
+    logger.warning(f'starting {function.__name__} on {key}')
+    while status is not True:
+        if token:
+            kwargs['token'] = token
+        token, status = function(**kwargs)
+        if not status:
+            time.sleep(4)  # wait 10 seconds for checkout to complete, then try again
+
+
+def parse_key(key):
+    try:
+        version = VERSION_REGEX.search(key).group(0)
+        uuid = UUID_REGEX.search(key).group(0)
+    except IndexError:
+        RuntimeError(f'unable to parse the key {key}')
+    return uuid, version
+
+
+@checkout.action("remove",
+                 arguments={"--replica": dict(choices=[r.name for r in Replica], required=True),
+                            "--keys": dict(nargs="+", help="keys to remove from checkout", required=False)})
+def remove(argv: typing.List[str], args: argparse.Namespace):
+    """
+    Remove a bundle from the checkout bucket
+    """
+    replica = Replica[args.replica]
+    handle = Config.get_blobstore_handle(replica)
+    bucket = replica.checkout_bucket
+    if args.keys:
+        for _key in args.keys:
+            if DSS_BUNDLE_KEY_REGEX.match(_key):
+                for key in handle.list(bucket, _key):  # handles checkout/bundle/*
+                    verify_delete(handle, bucket, key)
+                uuid, version = parse_key(key)
+                manifest = get_bundle_manifest(replica=replica, uuid=uuid, version=version)
+                if manifest is None:
+                    logger.warning(f"unable to locate manifest for fqid: {bucket}/{_key}")
+                    continue
+                for _files in manifest['files']:
+                    key = compose_blob_key(_files)
+                    verify_delete(handle, bucket, key)
+            elif FILE_PREFIX in _key:
+                # should handle other keys, files/blobs
+                file_metadata = verify_get(handle, replica.bucket, _key)
+                verify_delete(handle, bucket, key=compose_blob_key(file_metadata))
+
+
+@checkout.action("verify",
+                 arguments={"--replica": dict(choices=[r.name for r in Replica], required=True),
+                            "--keys": dict(nargs="+", help="keys to check the status of", required=False)})
+def verify(argv: typing.List[str], args: argparse.Namespace):
+    """
+    Verify that keys are in the checkout bucket
+    """
+    replica = Replica[args.replica]
+    handle = Config.get_blobstore_handle(replica)
+    bucket = replica.checkout_bucket
+    checkout_status = dict()
+    for _key in args.keys:
```

Comment:
Should handle the case of no keys being specified (`if args.keys:`) before iterating over args.keys, as done in the remove method above.

Alternatively, should `--keys` flag be changed to `required=True`?